base_layer/core/src/blocks/block_header.rs

```diff
@@ -243,12 +246,11 @@ impl From<NewBlockHeaderTemplate> for BlockHeader {
 
 impl Hashable for BlockHeader {
     fn hash(&self) -> Vec<u8> {
-        HashDigest::new()
-            .chain(self.merged_mining_hash())
-            .chain(self.pow.to_bytes())
-            .chain(self.nonce.to_le_bytes())
-            .finalize()
-            .to_vec()
+        let mut hasher = HashWriter::new(HashDigest::new());
+        self.merged_mining_hash().consensus_encode(&mut hasher).unwrap();
```

Comment:
This is not quite right because we still use a dynamic length for hashes, so consensus encoding will prepend 0x20 (32) to the encoding but hashes are fixed length so don't need that byte.

To deal with this, I suggest the following annoying 😛  code: (change var names as needed)

```rust
let hash = self.merge_mining_hash();
let mut hash_buf = [0u8; 32];
hash_buf.copy_from_slice(&hash[..32]);
hash_buf.consensus_encode(&mut writer).unwrap();
```

This will omit the redundant length byte from the encoding